In [1]:
import numpy as np

Задание 1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

$$cov_{XY} = M[(X - M(X))(Y - M(Y))] = M(XY) - M(X)M(Y)=\overline{X \cdot Y} - \overline{X} \cdot \overline{Y}$$

Зная ковариацию и среднее квадратичное отклонение каждого из двух признаков, можно вычислить коэффициент корреляции Пирсона:
$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [3]:
zp*ks

array([ 14035,  25830, 166060, 183800,  18360,  51730,  35262, 135300,
        89520,  91520])

$$cov_{XY}= \frac{14035+25830+166060+183800+18360+51730+35262+135300+89520+91520}{10}-\frac{35+45+190+200+40+70+54+150+120+110}{10} \cdot \frac{401+574+874+919+459+739+653+902+746+832}{10} $$

In [4]:
14035+25830+166060+183800+18360+51730+35262+135300+89520+91520

811417

In [5]:
35+45+190+200+40+70+54+150+120+110

1014

In [6]:
401+574+874+919+459+739+653+902+746+832

7099

In [7]:
1014*7099

7198386

In [8]:
8114170-7198386

915784

$$cov_{XY}= \frac{811417}{10}-\frac{1014}{10} \cdot \frac{7099}{10}=\frac{8114170}{100}-\frac{7198386}{100}=\frac{8114170-7198386}{100}=\frac{915784}{100}$$

In [9]:
cov_xy=np.mean(zp*ks)-np.mean(zp)*np.mean(ks) #проверка
cov_xy

9157.839999999997

Среднеквадратичное отклоение
$$\sigma = \sqrt{\frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n}}$$,
где $$\overline{X} = \frac{1014}{10}=101.4$$
$$\overline{Y} = \frac{7099}{10}=709.9$$

$$\sigma_{ZP} = \sqrt{\frac{(35-101.4)^2+(45-101.4)^2+(190-101.4)^2+(200-101.4)^2+(40-101.4)^2+(70-101.4)^2+(54-101.4)^2+(150-101.4)^2+(120-101.4)^2+(110-101.4)^2}{10}}=\sqrt{\frac{(-66.4)^2+(-56.4)^2+(88.6)^2+(98.6)^2+(-61.4)^2+(-31.4)^2+(-47.4)^2+(48.6)^2+(18.6)^2+(8.6)^2}{10}}=\frac{\sqrt{34946.399}}{3.16227}=\frac{186.939}{3.16227}=59.11548$$

In [10]:
np.std(zp)

59.115480206118605

In [11]:
np.sqrt(10)

3.1622776601683795

In [12]:
(66.4)**2+(56.4)**2+(88.6)**2+(98.6)**2+(61.4)**2+(31.4)**2+(47.4)**2+(48.6)**2+(18.6)**2+(8.6)**2

34946.399999999994

In [13]:
np.sqrt((66.4)**2+(56.4)**2+(88.6)**2+(98.6)**2+(61.4)**2+(31.4)**2+(47.4)**2+(48.6)**2+(18.6)**2+(8.6)**2)

186.93956242593487

$$\sigma_{KS} = \sqrt{\frac{(401-709.9)^2+(574-709.9)^2+(874-709.9)^2+(919-709.9)^2+(459-709.9)^2+(739-709.9)^2+(653-709.9)^2+(902-709.9)^2+(746-709.9)^2+(832-709.9)^2}{10}}=\sqrt{\frac{(-308.9)^2+(-135.9)^2+(164.1)^2+(209.1)^2+(-250.9)^2+(29.1)^2+(-56.9)^2+(192.1)^2+(36.1)^2+(122.1)^2}{10}}\frac{\sqrt{304688.899}}{3.16227}=\frac{551.98}{3.16227}=174.551$$

In [14]:
np.std(ks)

174.55340157098058

In [15]:
(308.9)**2+(135.9)**2+(164.1)**2+(209.1)**2+(250.9)**2+(29.1)**2+(56.9)**2+(192.1)**2+(36.1)**2+(122.1)**2

304688.89999999997

In [16]:
np.sqrt((308.9)**2+(135.9)**2+(164.1)**2+(209.1)**2+(250.9)**2+(29.1)**2+(56.9)**2+(192.1)**2+(36.1)**2+(122.1)**2)

551.9863222943119

Подставляем в формулу:
$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}=\frac{\frac{915784}{100}}{59.11548 \cdot 174.551}=\frac{915784}{100 \cdot 59.11548 \cdot 174.551}=\frac{915784}{1031871.49}=0.88749$$

In [17]:
cov_xy=np.cov(zp, ks, ddof=0) #проверка значения ковариации
cov_xy

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [18]:
np.corrcoef(zp, ks) #проверка коэффициента вариации Пирсона

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

в результате все сходится

Задание 2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

<br>$n= 10$
<br>$α=1-p=0.05$
<br>$p=0.95$

$\sigma$ не известна, не было анализа

Значит критерий $К$ равен критерию $Т$(Стьюдента) из таблицы Стьюдента.

$\overline{x}=\frac{131+125+115+122+131+115+107+99+125+111}{10}=118.1$

In [19]:
iq=np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
np.mean(iq)

118.1

Доверительный интервал ищем по формуле: 
$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}$$

где $c_{\gamma}=2.2622$ согласно таблице Стьюдента

Так как выборка $n < 100$ используем:
    $$\sigma_{несмещен} = \sqrt{\frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n-1}}$$

$$\sigma_{несмещен}=\sqrt{\frac{(131-118.1)^2+(125-118.1)^2+(115-118.1)^2+(122-118.1)^2+(131-118.1)^2+(115-118.1)^2+(107-118.1)^2+(99-118.1)^2+(125-118.1)^2+(111-118.1)^2}{9}}=\sqrt{\frac{(12.9)^2+(6.9)^2+(-3.1)^2+(3.9)^2+(12.9)^2+(-3.1)^2+(-11.1)^2+(-19.1)^2+(6.9)^2+(-7.1)^2}{9}}=\sqrt{\frac{(12.9)^2+(6.9)^2+(-3.1)^2+(3.9)^2+(12.9)^2+(-3.1)^2+(-11.1)^2+(-19.1)^2+(6.9)^2+(-7.1)^2}{9}}=\sqrt{\frac{1000.9}{9}}=\frac{31.637}{3}=10.545$$

In [20]:
(12.9)**2+(6.9)**2+(3.1)**2+(3.9)**2+(12.9)**2+(3.1)**2+(11.1)**2+(19.1)**2+(6.9)**2+(7.1)**2

1000.9000000000001

In [21]:
np.sqrt((12.9)**2+(6.9)**2+(3.1)**2+(3.9)**2+(12.9)**2+(3.1)**2+(11.1)**2+(19.1)**2+(6.9)**2+(7.1)**2)

31.637003650788426

In [23]:
sigma = np.std(iq, ddof=1) #проверка
sigma

10.54566788359614

$$T_{1,2} = 118.1 \pm \frac{10.545}{\sqrt{10}} \cdot 2.2622 =118.1 \pm \frac{10.545}{3.16227} \cdot 2.2622=118.1 \pm 7.54359$$

$$T_{1}= 118.1 + 7.54359 = 125.643$$
$$T_{2}= 118.1 - 7.54359 = 110.556$$

Доверительный интервал равен = $[110.556, 125.643]$

In [24]:
import scipy.stats as st

In [25]:
#Если мы работаем с небольшой выборкой (n <30), мы можем использовать функцию t.interval() 
#из библиотеки scipy.stats, чтобы вычислить доверительный интервал для среднего значения совокупности.
n=10
st.t.interval(confidence=0.95, df=n-1, loc=np.mean(iq), scale=st.sem(iq))  #проверка

(110.55608365158724, 125.64391634841274)

Задание 3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

$\sigma^2 = 25$ нам дано значение дисперсии
<br>$n=27$
<br>$p=0.95$
<br>$\overline{x}=174.2$

Известна  $𝜎$
значит критерий оценки $K$ является критерием Фишера  $𝑍$ из таблицы распределения Лапласа при $\frac{\alpha}{2}$ и $Z=-1.96$

Среднеквадратичное отклонение будет равно $\sqrt{\sigma^2}$, подставим все значения в формулу доверительно интервала:

$$T_{1,2} = 174.2 \pm \frac{\sqrt{25}}{\sqrt{27}} \cdot -1.96 =174.2 \pm \frac{5}{5.1961} \cdot -1.96=174.2 \pm 1.886$$

$$T_{1} = 174.2 - 1.886 = 172.314$$
$$T_{2} = 174.2 + 1.886 = 176.086$$

Доверительный интервал = $[172.314,176.086]$

In [26]:
np.sqrt(27)

5.196152422706632

In [29]:
(5/np.sqrt(27))*1.96

1.8860108793527774

In [30]:
n=27
sigma=5
st.norm.interval(confidence=0.95, loc=174.2, scale=(sigma/np.sqrt(n))) #проверка значений доверительного интервала

(172.3140237765397, 176.08597622346028)